agents/builtin/mysql/mysql.go

```diff
@@ -93,78 +125,184 @@ func (m *MySQL) Run(ctx context.Context) {
+		default:
+			l.Debugf("Normal query: %s.", currentESS)
+		}
+
+		mb := &qanpb.MetricsBucket{
+			DSchema:                pointer.GetString(currentESS.SchemaName), // TODO can it be NULL?
+			Queryid:                *currentESS.Digest,
+			Fingerprint:            *currentESS.DigestText,
+			NumQueries:             count,
+			NumQueriesWithErrors:   float32(currentESS.SumErrors - prevESS.SumErrors),
+			NumQueriesWithWarnings: float32(currentESS.SumWarnings - prevESS.SumWarnings),
+			MetricsSource:          qanpb.MetricsSource_MYSQL_PERFSCHEMA,
 		}
 
-		// From https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-11.html:
-		// > The Performance Schema could produce DIGEST_TEXT values with a trailing space.
-		// > This no longer occurs. (Bug #26908015)
-		*ess.DigestText = strings.TrimSpace(*ess.DigestText)
-
-		// TODO https://jira.percona.com/browse/PMM-3594
-		/*
-		   A ton of open questions. Should pmm-agent:
-		   * check that performance schema is enabled?
-		   * check that statement_digest consumer is enabled?
-		   * check/report the value of performance_schema_digests_size?
-		   * TRUNCATE events_statements_summary_by_digest before reading?
-		   * read events_statements_summary_by_digest every second? every 10 seconds? minute? other interval?
-		   * report rows with NULL digest?
-		   * get query by digest from events_statements_history_long?
-		   * check/report the value of performance_schema_events_statements_history_long_size?
-		   * set conditions for FIRST_SEEN / LAST_SEEN? what conditions?
-		   * group/aggregate results? how?
-		   * should github.com/percona/go-mysql/event be used?
-		*/
-
-		res.MetricsBucket = append(res.MetricsBucket, &qanpb.MetricsBucket{
-			Queryid:     *ess.Digest,
-			Fingerprint: *ess.DigestText,
-			DServer:     "TODO",
-			DDatabase:   "TODO",
-			DSchema:     "TODO",
-		})
+		for _, p := range []struct {
+			value float32  // result value: currentESS.SumXXX-prevESS.SumXXX
+			sum   *float32 // MetricsBucket.XXXSum field to write value
+			cnt   *float32 // MetricsBucket.XXXCnt field to write count
+		}{
+			// in order of events_statements_summary_by_digest columns
+
+			// convert picoseconds to seconds
+			{float32(currentESS.SumTimerWait-prevESS.SumTimerWait) / 1e12, &mb.MQueryTimeSum, &mb.MQueryTimeCnt},
+			{float32(currentESS.SumLockTime-prevESS.SumLockTime) / 1e12, &mb.MLockTimeSum, &mb.MLockTimeCnt},
+
+			{float32(currentESS.SumRowsAffected - prevESS.SumRowsAffected), &mb.MRowsAffectedSum, &mb.MRowsAffectedCnt},
+			{float32(currentESS.SumRowsSent - prevESS.SumRowsSent), &mb.MRowsSentSum, &mb.MRowsSentCnt},
+			{float32(currentESS.SumRowsExamined - prevESS.SumRowsExamined), &mb.MRowsExaminedSum, &mb.MRowsExaminedCnt},
+
+			{float32(currentESS.SumCreatedTmpDiskTables - prevESS.SumCreatedTmpDiskTables), &mb.MTmpDiskTablesSum, &mb.MTmpDiskTablesCnt},
+			{float32(currentESS.SumCreatedTmpTables - prevESS.SumCreatedTmpTables), &mb.MQueryTimeSum, &mb.MTmpTableCnt},
```

Comment:
I think here should be:
{float32(currentESS.SumCreatedTmpTables - prevESS.SumCreatedTmpTables), **&mb.MTmpTablesSum,** **&mb.MTmpTablesCnt**},

Reply:
Good catch! Fixed.